File: openapi-test-scala-server-spring-client-jdk/testapi/model/BaseEntity.scala
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.Optional

case class BaseEntity(
  @JsonProperty("createdAt") createdAt: OffsetDateTime,
  @JsonProperty("id") id: PetId,
  @JsonProperty("updatedAt") updatedAt: Optional[OffsetDateTime]
)

File: openapi-test-kotlin-server-quarkus-reactive-client-jdk/testapi/model/BaseEntity.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import java.time.OffsetDateTime

data class BaseEntity(
  @field:JsonProperty("createdAt") @NotNull val createdAt: OffsetDateTime,
  @field:JsonProperty("id") @NotNull val id: PetId,
  @field:JsonProperty("updatedAt") val updatedAt: OffsetDateTime?
)